mesonbuild/environment.py

```diff
@@ -693,28 +705,39 @@ def detect_static_linker(self, compiler):
             linker = self.cross_info.config['binaries']['ar']
             if isinstance(linker, str):
                 linker = [linker]
+            linkers = [linker]
         else:
             evar = 'AR'
             if evar in os.environ:
-                linker = shlex.split(os.environ[evar])
+                linkers = [shlex.split(os.environ[evar])]
             elif isinstance(compiler, VisualStudioCCompiler):
-                linker = [self.vs_static_linker]
+                linkers = [self.vs_static_linker]
+            elif isinstance(compiler, GnuCompiler):
+                # Use gcc-ar if available; needed for LTO
+                linkers = [self.gcc_static_linker, self.default_static_linker]
+            elif isinstance(compiler, ClangCompiler):
```

Comment:
__[flake8]__
* [F405] 'ClangCompiler' may be undefined, or defined from star imports: .compilers

<!-- Comment by [SideCI](https://sideci.com) -->